google/cloud/aiplatform/training_jobs.py

```diff
@@ -386,12 +417,47 @@ def _create_input_data_config(
 
         input_data_config = None
         if dataset:
-            # Create fraction split spec
-            fraction_split = gca_training_pipeline.FractionSplit(
-                training_fraction=training_fraction_split,
-                validation_fraction=validation_fraction_split,
-                test_fraction=test_fraction_split,
-            )
+            filter_split = None
+            if any(
+                [
+                    training_filter_split is not None,
+                    validation_filter_split is not None,
+                    test_filter_split is not None,
+                ]
+            ):
+                if all(
+                    [
+                        training_filter_split is not None,
+                        validation_filter_split is not None,
+                        test_filter_split is not None,
+                        dataset._gca_resource.metadata_schema_uri
+                        in (
+                            schema.dataset.metadata.image,
+                            schema.dataset.metadata.text,
+                            schema.dataset.metadata.video,
+                        ),
+                    ]
+                ):
+                    # Create filter split spec
+                    filter_split = gca_training_pipeline.FilterSplit(
+                        training_filter=training_filter_split,
+                        validation_filter=validation_filter_split,
+                        test_filter=test_filter_split,
+                    )
+                elif any(
+                    [
+                        training_filter_split is None,
+                        validation_filter_split is None,
+                        test_filter_split is None,
+                    ]
+                ):
+                    raise TypeError(
+                        "To use filter split, all of the following have to be provided: training_filter_split, validation_filter_split, test_filter_split"
+                    )
+                else:
+                    raise ValueError(
+                        "A filter split may only be used with datesets with DataItems"
+                    )
```

Comment:
I plan on writing a couple more tests to check the behaviour when multiple types of splits are passed. But before I do that, I'd like y'all's thoughts on whether this is what we'd want.